sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs

```diff
@@ -397,81 +655,26 @@ async Task<bool> OnBreakpointHit (SessionId sessionId, JObject args, Cancellatio
+
 			var callFrames = new List<object> ();
 			foreach (var frame in orig_callframes) {
 				var function_name = frame ["functionName"]?.Value<string> ();
 				var url = frame ["url"]?.Value<string> ();
-				if ("mono_wasm_fire_bp" == function_name || "_mono_wasm_fire_bp" == function_name) {
-					var frames = new List<Frame> ();
-					int frame_id = 0;
-					var the_mono_frames = res.Value? ["result"]? ["value"]? ["frames"]?.Values<JObject> ();
-
-					foreach (var mono_frame in the_mono_frames) {
-						++frame_id;
-						var il_pos = mono_frame ["il_pos"].Value<int> ();
-						var method_token = mono_frame ["method_token"].Value<uint> ();
-						var assembly_name = mono_frame ["assembly_name"].Value<string> ();
-
-						// This can be different than `method.Name`, like in case of generic methods
-						var method_name = mono_frame ["method_name"]?.Value<string> ();
-
-						var store = await LoadStore (sessionId, token);
-						var asm = store.GetAssemblyByName (assembly_name);
-						if (asm == null) {
-							Log ("info",$"Unable to find assembly: {assembly_name}");
-							continue;
-						}
-
-						var method = asm.GetMethodByToken (method_token);
-
-						if (method == null) {
-							Log ("info", $"Unable to find il offset: {il_pos} in method token: {method_token} assembly name: {assembly_name}");
-							continue;
-						}
-
-						var location = method?.GetLocationByIl (il_pos);
-
-						// When hitting a breakpoint on the "IncrementCount" method in the standard
-						// Blazor project template, one of the stack frames is inside mscorlib.dll
-						// and we get location==null for it. It will trigger a NullReferenceException
-						// if we don't skip over that stack frame.
-						if (location == null) {
-							continue;
-						}
-
-						Log ("info", $"frame il offset: {il_pos} method token: {method_token} assembly name: {assembly_name}");
-						Log ("info", $"\tmethod {method_name} location: {location}");
-						frames.Add (new Frame (method, location, frame_id-1));
-
-						callFrames.Add (new {
-							functionName = method_name,
-							callFrameId = $"dotnet:scope:{frame_id-1}",
-							functionLocation = method.StartLocation.AsLocation (),
-
-							location = location.AsLocation (),
-
-							url = store.ToUrl (location),
-
-							scopeChain = new [] {
-								new {
-									type = "local",
-									@object = new {
-										@type = "object",
-										className = "Object",
-										description = "Object",
-										objectId = $"dotnet:scope:{frame_id-1}",
-									},
-									name = method_name,
-									startLocation = method.StartLocation.AsLocation (),
-									endLocation = method.EndLocation.AsLocation (),
-								}}
-						});
-
-						context.CallStack = frames;
-
-					}
+				if (function_name == "mono_wasm_fire_exc" || function_name == "_mono_wasm_fire_exc") {
+					Log ("info", $"exception frame");
+					exception_data = await HandleExceptionFrame (sessionId, frame, token);
```

Comment:
We could in theory, but I don't see much value in doing so, if any at all because it'd be a very obscure situation that should not even happen.  In fact, we might even just hard throw instead of returning null.